[PepperBridge] fix and discussion

sysadminstory opened the pull request:
Some expired Deal does not have any URL and generated some PHP Notice. Now they won't generate any Notice.
Expired deal have a different format for the title : this is now handled too.

The "Most commented" sort order was removed from the Pepper Website, so it is removed from the bridge too.
Bridge using this sort order will break, but this is intended (as they are broken)

A new context allows to monitor a Discussion for new comments, and if needed, only show comment containing an URL.
This allows to close #2354

sysadminstory committed:
[PepperBridge] Fix some Notice for expired items

Some expired iteams are not identical to normal items for the title and
URI, so they got a special handling to remove PHP Notice.

sysadminstory committed:
[PepperBridge] Remove "most commented" sorting option

The "most commented" sorting option was removed and show now some forum.
So it was removed.

sysadminstory committed:
[PepperBridge] Add Discussion monotoring mode

With DealabsBridge, MydealsBridge and HotUKDealsBridge, you can monitor
a discussion for new comments, excluding or not post without URL

sysadminstory committed:
[PepperBridge] Fix checkboc attribute

Remove unsupported 'required' attribute for checkboxes

em92 commented on bridges/DealabsBridge.php:
```diff
@@ -2081,6 +2101,128 @@ protected function collectDeals($url){
+			'on BadgeLevel{key name description}fragment paginationFields on Pagination{count current last',
+			' next previous size order}'
+		);
+		$graphqlString = implode('', $graphqlArray);
```
How about using multline php string?

sysadminstory replied:
Hmm, do you mean using the Heredoc syntax :
https://www.php.net/manual/en/language.types.string.php#language.types.string.syntax.heredoc
?

Cutting this string to words is a bit of nightmare, but I can do it of needed !

em92 replied:
> do you mean using the Heredoc syntax :

Yes, I mean using heredoc syntax.

> Cutting this string to words is a bit of nightmare, but I can do it of needed !

The nightmare is reading this graphql query.

As an alternative, if you copied that graphql query from some code as is, you could just point where to get this string.

sysadminstory replied:
Ok, I understand :)
I didn't write this GraphQL Query : I discovered the name while writing the bridge !

This query was extracted using the network tab of Firefox while visiting the website !

I used nowdoc (because some content looked like PHP in it.

em92 commented on bridges/DealabsBridge.php:
```diff
@@ -2081,6 +2101,128 @@ protected function collectDeals($url){
+				CURLOPT_POSTFIELDS => $queryJSON
+				);
+		$json = getContents($url, $header, $opts)
+			or returnServerError($this->i8n('request-error'));
```
https://github.com/RSS-Bridge/rss-bridge/pull/2398 :)

sysadminstory replied:
:heavy_check_mark: :)

sysadminstory committed:
[PepperBridge] fix and discussion

Remove redundant "or returnServerError" after getContents

sysadminstory committed:
[PepperBridge] fix and discussion

The GraphQL was converted to a nowdoc string to get it more readable.
This GraphQL query was extracted from the website, this is not part of
the code of this Bridge.

sysadminstory committed:
[PepperBridge] fix and discussion

Coding policy fix !
